delivery_transsmart/models/res_config.py

```diff
@@ -106,83 +106,150 @@ def get_transsmart_service(self):
         return self.env['delivery.web.service'].browse([wst])
 
     def get_transsmart_carrier_tag(self):
-        return self.env['ir.model.data'].get_object('delivery_transsmart', 'res_partner_category_transsmart_carrier')        
+        return self.env.ref(
+                'delivery_transsmart.res_partner_category_transsmart_carrier'
```

Comment:
Fix indent. Last ).id can go on line before, or if you prefer to keep it on its own line, the ) should be in the same column as the r of return.